component-api/src/main/java/org/talend/sdk/component/api/configuration/ui/widget/ColumnList.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (C) 2006-2023 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.api.configuration.ui.widget;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.talend.sdk.component.api.configuration.ui.meta.Ui;
+import org.talend.sdk.component.api.meta.Documentation;
+
+@Ui
+@Documentation("Mark a ui option's suggestion values are filled by current component schema or pre connection line's schema auto, only for studio")
+@Retention(RUNTIME)
+@Target({ PARAMETER, FIELD })
+public @interface ColumnList {
```

Comment:
Isn't `ColumnReference`  a more accurate name?
```suggestion
public @interface ColumnReference {
```

Reply:
or `FieldReference`?
```suggestion
public @interface FieldReference {
```

Reply:
this ui is a closedlist widget, which can choose one column from talend schema, so i think XXXList is better. And "Column" is more like a ui thing than Field.